estela-api/api/views/auth.py

```diff
@@ -154,6 +154,50 @@ def get_queryset(self):
             return self.queryset.filter(username=self.request.user.username)
         return self.queryset
 
+    @swagger_auto_schema(
+        responses={status.HTTP_200_OK: UserProfileSerializer()},
+    )
+    def retrieve(self, request, *args, **kwargs):
+        user: User = request.user
+        requested_user: User = User.objects.filter(username=kwargs["username"]).first()
+
+        if requested_user == None:
+            return Response(
+                data={"error": "This user doesn't exist in estela."},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        if user != requested_user:
```

Comment:
It's necessary this verification? permission_classes could handle it.

Reply:
I was researching about this but couldn't find any integrated solution in rest_framework to validate this, so I put this validation explicit in the code.

Reply:
nice!